lib/Dialect/FIRRTL/Transforms/SFCCompat.cpp

```diff
@@ -46,13 +46,22 @@ void SFCCompatPass::runOnOperation() {
   bool madeModifications = false;
   SmallVector<InvalidValueOp> invalidOps;
 
-  bool fullAsyncResetExists = false;
-  AnnotationSet::removePortAnnotations(
+  auto fullAsyncResetAttr =
+      StringAttr::get(&getContext(), fullAsyncResetAnnoClass);
+  auto isFullAsyncResetAnno = [fullAsyncResetAttr](Annotation anno) {
+    return anno.getClassAttr() == fullAsyncResetAttr;
+  };
+  bool fullAsyncResetExists = AnnotationSet::removePortAnnotations(
       getOperation(), [&](unsigned argNum, Annotation anno) {
-        if (!anno.isClass(fullAsyncResetAnnoClass))
-          return false;
-        return fullAsyncResetExists = true;
+        return isFullAsyncResetAnno(anno);
       });
+  if (!fullAsyncResetExists) {
+    getOperation()->walk([&](Operation *op) {
```

Comment:
This walk is unfortunate.

It's also good to be sure to fully remove these annotations.

Reply:
Is it guaranteed that there is no isFullAsyncResetAnno when isFullAsyncResetAnno is on ports?

Reply:
It's checked by a previous pass but that's not great, good question.  Might be better to just walk it all here anyway.

Reply:
Fixed to always walk, thanks!